api/v1beta1/watcherapi_types.go

```diff
@@ -17,25 +17,28 @@ limitations under the License.
 package v1beta1
 
 import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // WatcherAPISpec defines the desired state of WatcherAPI
 type WatcherAPISpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of WatcherAPI. Edit watcherapi_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	WatcherTemplate `json:",inline"`
```

Comment:
this might just be githubs sucky ui but where is thie type defined?

Reply:
lookign more closely this is incrrect 

https://github.com/openstack-k8s-operators/watcher-operator/blob/main/api/v1beta1/common_types.go#L20

that is the template for the Watcher top-level CRD

you should not be embding that in the watcherAPISpec.

the WatcherTemplate define the public API presented to the end user when the top-level watcher operator is integrated with the opesntack operator.

the way it will be used si the oepnstack operator will embde the WatcherTemplate in the OpenstackContolePlane CR it will then use it with a defaultign fuctionin the API module to create a concrete instance of the watcher CRD.

the watcher controller will reconsile the watcher CR and generate a WatcherAPI CR which is an instance of the WathcerAPISpec.

the WatcherTemplate will contain field related to the applier/decision engine that are not approapte to embded in the WatcherAPISpec

you should directly copy the required fields or create a separate shareable class for the existing fields that you sue both here and in the watcherTemplate.

Reply:
ack, didn't know that about the WatcherTemplate, thanks for the explanation, I'll duplicate the fields and we can see if it's worth it to refactor into a shareable struct in the future

Reply:
TBH, when I created WatcherTemplate I was thinking on it to be the common shareable part of the SPECs for all the watcher CRs, but I think on it being used by the uppler level openstack operator. 

it make sense to create another lower struct with the common part.

Reply:
by the way this is why i want to avoid the term template for the base class

https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_types.go#L772-L788


for each service section we have a field called template

```Template *barbicanv1.BarbicanSpecCore `json:"template,omitempty"```

that is an instance of the <service>CoreSpec class.

Reply:
I'm perfectly fine  with this. Actually, I think using the term "watcherTemplate" to describe what will be used by openstack operator to create a Watcher makes sense. Also, given that there is no consistency accross operators, I'm flexible in using whatever convention we decide as soon as we are aligned on it.